src/ui/Distance/DistanceResources.ts

```diff
@@ -0,0 +1,245 @@
+     */
+    distanceField: ComponentOptions.buildStringOption({
+      required: true
+    }),
+    /**
+     * Specifies the field that contains the latitude value.
+     *
+     * Specifying a value for this option is required for the `DistanceResources` component to work.
+     */
+    latitudeField: ComponentOptions.buildStringOption({
+      required: true
+    }),
+    /**
+     * Specifies the field that contains the longitude value.
+     *
+     * Specifying a value for this option is required for the `DistanceResources` component to work.
+     */
+    longitudeField: ComponentOptions.buildStringOption({
+      required: true
+    }),
+    /**
+     * The conversion factor to use for the distance according to the base unit, in meters.
+     *
+     * If you want to have Kilometers, you should set 1000 as the value.
+     * If you want to have your distance in miles, you should set 1610 as the value, since a mile is approximately 1610 meters.
+     *
+     * The default value is `1000`.
+     */
+    unitConversionFactor: ComponentOptions.buildNumberOption({
+      defaultValue: 1000,
+      validator: value => {
+        return !!value && value > 0;
+      }
+    }),
+    /**
+     * The CSS class for components that needs to be reenabled when the distance is provided.
+     * 
+     * The default value is `coveo-distance-disabled`.
+     */
+    disabledDistanceCssClass: ComponentOptions.buildStringOption({
+      defaultValue: 'coveo-distance-disabled'
+    })
+  };
+
+  /**
+   * Creates a new `DistanceResources` component.
+   * @param element The HTMLElement on which to instantiate the component.
+   * @param options The options for the `DistanceResources` component.
+   * @param bindings The bindings that the component requires to function normally. If not set, these will be
+   * automatically resolved (with a slower execution time).
+   */
+  constructor(public element: HTMLElement, public options: IDistanceOptions, public bindings: IComponentBindings) {
+    super(element, DistanceResources.ID, bindings);
+
+    this.options = ComponentOptions.initComponentOptions(element, DistanceResources, options);
+    this.bind.onRootElement(InitializationEvents.afterComponentsInitialization, () => this.onAfterComponentsInitialization());
+  }
+
+  /**
+   * Override the current position with the provided values.
+   *
+   * Triggers a query automatically.
+   * @param latitude The latitude to set.
+   * @param longitude The longitude to set.
+   */
+  public setPosition(latitude: number, longitude: number): void {
+    this.latitude = latitude;
+    this.longitude = longitude;
+
+    const args: IPositionResolvedEventArgs = {
+      position: {
+        lat: latitude,
+        long: longitude
+      }
+    };
+
+    this.bind.trigger(this.element, DistanceEvents.onPositionResolved, args);
+    this.registerDistanceQuery();
+    this.queryController.executeQuery();
```

Comment:
We'll need the usage analytics plugged in every time this component executes a query.